docs/internals/details.rst

```diff
@@ -0,0 +1,46 @@
+Implementation Details
```

Comment:
In my opinion, the section title "Implementation Details" doesn't accurately describe what's being discussed here.  It's a Stan implementation detail, but not a BridgeStan implementation detail.  No doubt, the consequences of this Stan implementation detail affect BridgeStan.

Reply:
Very happy to put it somewhere else or title it something different, I just did not know where to fit it in the current docs so I invented a new page

Reply:
I went with "Stan implementation details"